### PR TITLE
Check leap year and number of days in each month

### PR DESCRIPTION
**Pull request by jasson99**
Check leap year and number of days in each month

### CODE REVIEW

File: bsdate.go [hunk 1]
```diff
@@ -194,11 +194,27 @@ func NewFromGregorian(gregorianDay, gregorianMonth, gregorianYear int) (Date, er
 	var bsYear = gregorianYear + 56         //first rough calculation, might become 57 later
 	var bsMonth = 9                         //Jan 1 always fall in BS month Paush which is the 9th month
 	var daysSinceJanFirstToEndOfBsMonth int //days calculated from 1st Jan till the end of the actual BS month,
-	                                        // we use this value to check if the gregorian Date is in the actual BS month
+	// we use this value to check if the gregorian Date is in the actual BS month
```
Comment (individual-it) [inline bsdate.go]: why this change?

File: bsdate.go [hunk 2]
```diff
@@ -266,6 +282,9 @@ func (d date) isValid() bool {
 	if d.Day <= 0 || d.Day > 32 || d.Month <= 0 || d.Month > 12 || d.Year <= 0 {
 		return false
 	}
+	if reflect.TypeOf(d.Day) == reflect.TypeOf("string") || reflect.TypeOf(d.Month) == reflect.TypeOf("string") || reflect.TypeOf(d.Year) == reflect.TypeOf("string") {
```
Comment (individual-it) [inline bsdate.go]: long line
this change is not mentioned in the description

File: bsdate.go [hunk 3]
```diff
@@ -194,11 +194,27 @@ func NewFromGregorian(gregorianDay, gregorianMonth, gregorianYear int) (Date, er
...
 	var bsMonth = 9                         //Jan 1 always fall in BS month Paush which is the 9th month
 	var daysSinceJanFirstToEndOfBsMonth int //days calculated from 1st Jan till the end of the actual BS month,
-	                                        // we use this value to check if the gregorian Date is in the actual BS month
+	// we use this value to check if the gregorian Date is in the actual BS month
 
 	if _, ok := calendardata[bsYear]; !ok {
 		return nil, errors.New("cannot convert date, missing data")
 	}
+	// Months with 31 days
+	if gregorianMonth == 2 || gregorianMonth == 4 || gregorianMonth == 6 || gregorianMonth == 9 || gregorianMonth == 11  {
```
Comment (individual-it) [inline bsdate.go]: long line

File: bsdate.go [hunk 4]
```diff
@@ -194,11 +194,27 @@ func NewFromGregorian(gregorianDay, gregorianMonth, gregorianYear int) (Date, er
...
+		if gregorianDay > 30 {
+			return nil, errors.New("cannot convert date, missing data")
+		}
+	}
+	// is the year leap year? Leap year has 29 days in february
+	if (gregorianYear % 4 == 0 &&  gregorianYear % 100 != 0) || gregorianYear % 400 == 0 {
+				if gregorianMonth == 2 && gregorianDay > 29 {
+					return nil, errors.New("cannot convert date, missing data")
+				}
```
Comment (individual-it) [inline bsdate.go]: wrong indent

File: bsdate.go [hunk 5]
```diff
@@ -277,13 +296,12 @@ func (d date) isValid() bool {
 	return true
 }
 
-func (d date) GetGregorianDate() (time.Time, error)  {
+func (d date) GetGregorianDate() (time.Time, error) {
 	var daysAfterJanFirstOfGregorianYear = 0 //we will add all the days that went by since the 1st.
-	                                         //January and then we can get the gregorian Date
+	//January and then we can get the gregorian Date
```
Comment (individual-it) [inline bsdate.go]: why this change?

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — CHANGES_REQUESTED** [review]
please also add extra tests for your changes

File: bsdate.go (hunk 1 above)
Comment (jasson99) [reply bsdate.go]: fixed. It happened when I used the auto indent

File: bsdate.go (hunk 2 above)
Comment (jasson99) [reply bsdate.go]: fixed

File: bsdate.go (hunk 4 above)
Comment (jasson99) [reply bsdate.go]: fixed

File: bsdate.go (hunk 5 above)
Comment (jasson99) [reply bsdate.go]: fixed

File: bsdate.go (hunk 3 above)
Comment (jasson99) [reply bsdate.go]: fixed

**Comment by jasson99** [comment]
It was made from forked repo
